webv2/web.go

```diff
@@ -1798,6 +1909,76 @@ func getSourceAndTargetProfiles(sessionState *session.SessionState, details migr
 	return sourceProfile, targetProfile, ioHelper, dbName, nil
 }
 
+func getSourceProfileStringForShardedMigrations(sessionState *session.SessionState, details migrationDetails) (string, error) {
+	fileName := "HB-" + uuid.New().String() + "-sharding.cfg"
+	if details.MigrationType != helpers.LOW_DOWNTIME_MIGRATION {
+		err := createConfigFileForShardedBulkMigration(sessionState, details, fileName)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("config=%v", fileName), nil
+	} else if details.MigrationType == helpers.LOW_DOWNTIME_MIGRATION {
+		err := createConfigFileForShardedDataflowMigration(sessionState, details, fileName)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("config=%v", fileName), nil
+	} else {
+		return "", fmt.Errorf("this migration type is not implemented yet")
```

Comment:
will this condition ever be reached ?

Reply:
1. This is to eagerly throw an error to identify point of integration when adding new migration modes (DMS, Dataproc). 
3. A UI bug can also cause this condition to occur since input to `web.go` is passed from UI.
2. Added this so that the code exits cleanly instead of somewhere deeper down the stack.

Reply:
but if checks for 
type=low_downtime
else if
type!=low_downtime

this is basically the whole problem space. Any condition after this will not come up